File: XUCore/additions/NSMutableDictionaryAdditions.m
// 
// NSMutableDictionaryAdditions.m
// 
// Created by Charlie Monroe
// 
// Copyright (c) 2010-13 Fuel Collective, LLC. All rights reserved.
// 


#import "NSMutableDictionaryAdditions.h"

@implementation NSMutableDictionary (NSMutableDictionaryAdditions)

-(void)setBool:(BOOL)aBool forKey:(id)key{
	[self setObject:@(aBool) forKey:key];
}
-(void)setFloat:(float)aFloat forKey:(id)key{
	[self setObject:@(aFloat) forKey:key];
}
-(void)setInt:(int)anInt forKey:(id)key{
	[self setObject:@(anInt) forKey:key];
}
-(void)setUnsignedInt:(unsigned int)anInt forKey:(id)key{
	[self setObject:@(anInt) forKey:key];
}
-(void)setUnsignedShort:(unsigned short)aShort forKey:(id)key{
	[self setObject:@(aShort) forKey:key];
}
-(BOOL)setObjectConditionally:(id)object forKey:(id)key{
	if (object != nil){
		[self setObject:object forKey:key];
		return YES;
	}
	return NO;
}

@end

File: XUCore/additions/NSMutableDictionaryAdditions.swift
//
//  NSMutableDictionaryAdditions.swift
//  XUCore
//
//  Created by Charlie Monroe on 11/11/15.
//  Copyright © 2015 Charlie Monroe Software. All rights reserved.
//

import Foundation

public extension NSMutableDictionary {
	
	public func setBool(_ aBool: Bool, forKey key: Key) {
		self[key] = aBool
	}
	public func setFloat(_ aFloat: Float, forKey key: Key) {
		self[key] = aFloat
	}
	public func setInt(_ anInt: Int32, forKey key: Key) {
		self[key] = NSNumber(value: anInt as Int32)
	}
	public func setUnsignedInt(_ anInt: UInt32, forKey key: Key) {
		self[key] = NSNumber(value: anInt as UInt32)
	}
	public func setUnsignedShort(_ aShort: UInt16, forKey key: Key) {
		self[key] = NSNumber(value: aShort as UInt16)
	}
	public func setObjectConditionally(_ object: AnyObject?, forKey key: Key) -> Bool {
		if object != nil {
			self[key] = object!
			return true
		}
		return false
	}
	
}


